Update dataset documentation (#1044)
timescale/promscale: docs/dataset.md

# Dataset configuration format

> Promscale stores some configuration information in the Postgres database which it is connected to. We call this configuration the `Promscale dataset configuration`. Promscale accepts a configuration to set the dataset values. This document describes its format and mechanics.

Reviewer: ```suggestion
Promscale stores some configuration information in the Postgres database which it is connected to. We call this configuration the _Promscale dataset configuration_. Promscale accepts an option to set the dataset values. This document describes its format and mechanics.
```

Author: 👍
